assignment2/binarytree.py

```diff
@@ -0,0 +1,144 @@
+from random import randint
+from node import *
+
+class BinaryTree:
+    """Implementation of a Binary Tree."""
+
+    def __init__(self, rootKey = None):
+        self.root = None
+        self.size = 0
+        if rootKey is not None:
+            self.root = Node(rootKey)
+            self.size += 1
+
+    def getRoot(self):
+        """Return a Node that represents the root of the tree."""
+        return self.root
+
+    def getSize(self):
+        """Return the size of the tree."""
+        return self.size
+
+    def find(self, key):
+        """Does tree contains a node with such key.
+
+        Args:
+            key: an integer, key which should be found in the tree.
+
+        Returns:
+            a bool, True if the tree contains such node, and False otherwise.
+        """
+        return self._find(key, self.root)
+
+    def _find(self, key, cur):
```

Comment:
Fair enough: This is needed only within the class, so put it within the class makes sense.